src/Idris/Parser/Expr.hs

```diff
@@ -678,16 +679,17 @@ app syn = do f <- simpleExpr syn
                                f
                                (PMatchApp fc ff)
                                (PRef fc [] (sMN 0 "match")))
-              <?> "matching application expression") <|> (do
-             fc <- getFC
-             i <- get
-             args <- many (do notEndApp; arg syn)
-             wargs <- if withAppAllowed syn && not (inPattern syn)
-                         then many (do notEndApp; reservedOp "|"; expr' syn)
-                         else return []
-             case args of
-               [] -> return f
-               _  -> return (withApp fc (flattenFromInt fc f args) wargs))
+                   <?> "matching application expression") <|>
+               (do
```

Comment:
Why is this do alone on a line instead of together with the `fc <- getFC` step? That would more closely match the style in Idris.